apisix/cli/schema.lua

```diff
@@ -302,110 +683,87 @@ local config_schema = {
                                 type = "integer"
                             },
                             http_request_phase = {
-                                enum = {"access", "rewrite"},
+                                enum = { "access", "rewrite" },
                                 default = "access",
                             },
                         },
-                        required = {"name", "file", "priority"}
+                        required = { "name", "file", "priority" }
                     }
                 }
             }
         },
+        plugin_attr = {
+            type = "object"
+        },
         deployment = {
             type = "object",
             properties = {
                 role = {
-                    enum = {"traditional", "control_plane", "data_plane", "standalone"},
+                    enum = { "traditional", "control_plane", "data_plane", "standalone" },
                     default = "traditional"
-                }
+                },
             },
-        },
-    },
-    required = {"apisix", "deployment"},
-}
-
-local admin_schema = {
-    type = "object",
-    properties = {
-        admin_key = {
-            type = "array",
-            properties = {
-                items = {
-                    properties = {
-                        name = {type = "string"},
-                        key = {type = "string"},
-                        role = {type = "string"},
+            dependencies = {
+                role = {
+                    oneOf = {
+                        {
```

Comment:
@pottekkat Will you make sure that this dependency schema fits right with this[1] yaml snippet here?
1. https://github.com/apache/apisix/blob/88406dceab26ed7f25b663088b0b5e760bd45a38/t/cli/test_validate_config.sh#L210C1-L210C1